src/pCN.jl

```diff
@@ -0,0 +1,18 @@
+# Preconditioned Crank–Nicolson algorithm tools
+function pCN(source::AbstractNoiseProcess{T,N,Vector{T2},inplace}, ρ;
+              reset=true,reverse=false,indx=nothing) where {T,N,T2,inplace}
+
+  # generate new Wiener process similar to the one in source
```

Comment:
Shouldn't any `NP <: AbstractNoiseProcess` give a function to create an independent copy?

Reply:
like `deepcopy(source)` as in the new version for `pCN`?

Reply:
That's a good question, is there another case that modifies like this? I think NoiseWrapper actually does?

Reply:
Yes, so `NoiseWrapper` does modify the source that it takes as input. That could be bad if several similar correlated Wiener processes should be simulated with `pcN!`.  The `deepcopy(source)` call in `pCN`  should avoid this issue.  (I think we could alternatively also define `NoiseWrapper` and `NoiseWrapper!` but this would mean that the previous behaviour of `NoiseWrapper` changes, which sounds like a bigger change.)
 
Besides copying the source, using `save_everystep=false` could maybe also work (haven't tested that yet but I thought that doing a full copy is safer anyway.)

> is there another case that modifies like this? 
Does case refer to another `NoiseProcess` as the source like `NoiseGrid`? I would think that `deepcopy` is good for all of them.